Adding Tokens Overview (wayfinding) (#570)
stellar/stellar-docs: docs/smart-contracts/tokens/wayfinding.mdx

---
sidebar_position: 5
title: Tokens Overview
description: Choose between the Stellar Asset Contract and Custom Tokens
---

> Tokens exist in two forms on Stellar: 1) assets issued by Stellar accounts (Stellar assets can have a [Stellar Asset Contract (SAC)](./stellar-asset-contract.mdx) implementation), and 2) as a [custom token](./token-interface.mdx) created by a smart contract.

Reviewer: I think it might be easier to read if these 1 and 2 points are on new lines in a number list form.

> Tokens exist in two forms on Stellar: 1) assets issued by Stellar accounts (Stellar assets can have a [Stellar Asset Contract (SAC)](./stellar-asset-contract.mdx) implementation), and 2) as a [custom token](./token-interface.mdx) created by a smart contract.

Reviewer: Suggest:
```suggestion
Tokens exist in two forms on Stellar:
1. Assets issued by Stellar accounts and their built-in [Stellar Asset Contract (SAC)](./stellar-asset-contract.mdx) implementation)
2. [Custom tokens](./token-interface.mdx) issued by a deployed WASM contract.
```

> Several factors (such as project-specific requirements and network capabilities) can help you determine whether to issue an asset on Stellar or create a custom token with a smart contract for your project.

Reviewer: There's a lot of brackets in the overview, and the contents of these brackets while useful are not critical. Suggest:
```suggestion
Several factors can help you determine whether to issue an asset on Stellar or create a custom token with a smart contract for your project.
```

However:

### TL;DR

> If possible, we recommend issuing a Stellar asset and (if needed) using the SAC to interact with that asset in smart contracts. More on why below.

Reviewer: Suggest:
```suggestion
If possible, we recommend issuing a Stellar asset and using the SAC to interact with that asset in smart contracts or to send to contract addresses. More on why below.
```

## Issuing assets on Stellar

> Stellar is first-class in asset tokenization — issuing an asset can be done in [just one single transaction](https://developers.stellar.org/docs).

Reviewer: Is the goal to say Stellar is the best, or that Stellar has first-class support for assets? I think at the moment this does a bit of both, but maybe stronger with the below.

Also, one transaction to issue a asset is maybe a little misleading. In most cases you need to set things up, but I think the fact is that you don't have to write a contract.

I don't feel strongly about this if you prefer the existing wording here.

```suggestion
Stellar has first-class support for asset tokenization — issuing an asset can be done using a [built-in transaction](https://developers.stellar.org/docs) without the development of a smart contract.
```

> Stellar’s transactions are fast and cost-effective, making the network great for remittances and micropayments. It also has built-in features for compliance, asset management, and auditing. If you are just looking to perform simple transfers, issuing assets on Stellar has all the needed capabilities.

Reviewer: ```suggestion
Stellar’s transactions are fast and cost-effective, making the network great for remittances and micropayments. It also has built-in features for compliance, asset management, and auditing. If you are looking to perform transfers of value, issuing assets on Stellar has all the needed capabilities.
```

> Stellar assets:
> 
> Are already widely supported in the Stellar ecosystem and other ecosystems (such as Stellar USDC on Coinbase). Benefit from [anchors](../../learn/encyclopedia/anchors.mdx), the bridges between the Stellar network and traditional financial systems. Explore the global [Stellar anchor directory](https://anchors.stellar.org/) for further details. Give the issuer granular control over asset management with features that allow the issuer to [publish asset information](../../issuing-assets/publishing-asset-info.mdx), [ensure compliance](../../issuing-assets/anatomy-of-an-asset.mdx#compliance), and [regulate asset supply](../../issuing-assets/control-asset-access.mdx#limiting-the-supply-of-an-asset).

Reviewer: ```suggestion
Stellar assets are widely supported in the Stellar ecosystem and other ecosystems (such as Stellar USDC on Coinbase), and benefit from [anchors](../../learn/encyclopedia/anchors.mdx), the bridges between the Stellar network and traditional financial systems.

Explore the global [Stellar anchor directory](https://anchors.stellar.org/) for further details. Give the issuer granular control over asset management with features that allow the issuer to [publish asset information](../../issuing-assets/publishing-asset-info.mdx), [ensure compliance](../../issuing-assets/anatomy-of-an-asset.mdx#compliance), and [regulate asset supply](../../issuing-assets/control-asset-access.mdx#limiting-the-supply-of-an-asset).
```

Assets issued on the Stellar network are accessible to smart contracts with the use of that asset’s Stellar Asset Contract (SAC).

### Stellar Asset Contract

> The Stellar Asset Contract (SAC) is compiled into the protocol layer and allows smart contracts to interact with assets issued on Stellar. Every Stellar asset has reserved a SAC on the network that can be deployed by anyone who wants to interact with the asset from a contract. The SAC has access to all account (for XLM) and trustline (for all other assets) balances as well as smart contract token balances.

Reviewer: Couple tweaks:
- Use builtin for built in contracts. It's the vocab we've been trying to use.
- The last sentence is a little confusing, I thought it was trying to say something else.

```suggestion
The Stellar Asset Contract (SAC) is built into the protocol layer and allows smart contracts to interact with assets issued on Stellar. Every Stellar asset, including the native asset XLM, has reserved a SAC on the network that can be deployed by anyone who wants to interact with the asset from a contract.
```

Reviewer 2: I'd also change the "reserved" language as well. Maybe something like "A single instance of the SAC can be deployed for every Stellar asset by anyone who wants to interact with the asset from a contract"

Reviewer 2: I really like the "two interfaces to the same thing" analogy between non-smart-contract operations and the SAC. Maybe we can add this point here as well.

Read more about the SAC [here](./stellar-asset-contract.mdx).

Learn how to deploy a Stellar Asset Contract for an asset in [this How-To Guide](../../smart-contracts/guides/cli/deploy-stellar-asset-contract.mdx).

**Benefits of the SAC:**

> - Compatibility: Stellar ecosystem products (such as Stellar wallets) and other ecosystem products (such as exchanges) already know how to interact with Stellar assets, making the SAC more interoperable.

Reviewer: This is a Pro of Stellar Assets, not of the SAC itself. The SAC is just a means to an end.

I think this section could focus and frame the benefits of Stellar assets vs custom tokens.

Author: Ok I changed the wording around a bit to emphasize the importance of Stellar assets v. custom tokens

- Cost and resource efficiency: the SAC is built into the protocol-level code that already runs the network. So, there is no need to spin up a Wasm contract’s VM to invoke a function. Each function within the SAC will be more resource-efficient than its custom-coded counterpart.

> - Less work: you don’t have to write an entirely new contract. A Stellar asset’s SAC already exists on the network and just needs to be deployed to be used.

Reviewer: We should add:
- Customization: Stellar assets have the following customizations: x, y, z, admin addresses can be contracts, etc.

Author: I added some things here. Mostly related to Stellar assets, not the SAC. LMK what you think!

**Downside of the SAC:**

> - You cannot customize the SAC (with one exception, explained below). If you’re looking to use assets in a way not supported by the SAC, you can create your own custom smart contract token using the token interface. **Exception:** the asset issuer is automatically the admin for that asset’s SAC. However, the SAC allows the asset issuer to set a different admin address. Making this admin another smart contract means that you can add a bit of customization and logic to how the asset will authorize balances and trustlines, mint tokens, etc.

Reviewer: ```suggestion
- Other than the customization noted above, it is not possible to modify the behavior of Stellar Assets or their SAC. If you’re looking to use assets in a way not supported by Stellar assets, you can create your own custom smart contract token using the token interface and all applications that interact with tokens using the token interface will be able to interact with the custom token.
```

## Custom tokens

> If you have a unique use case where the SAC's predefined capabilities do not work for you, you can create a custom token using the [token interface](./token-interface.mdx). The token interface is a software interface that specifies what it means to be a smart contract token. The user can implement their own token contract that uses all the functions of this interface, and all tokens that implement this interface are interoperable with each other. The token interface executes on the smart contract layer instead of the protocol layer.

Reviewer: ```suggestion
If you have a unique use case where the SAC's capabilities are not sufficient, you can create a custom token that implements the [token interface](./token-interface.mdx). The token interface specifies the functions and events a contract must implement to be compatible with applications that use tokens.

The SAC also implements the token interface and applications that interoperate with the token interface can seamlessly interact with Stellar assets and custom tokens.
```

:::note

> Tokens created with smart contracts cannot interact with Stellar assets unless that Stellar asset has a deployed SAC.

Reviewer: ```suggestion
Tokens created with smart contracts cannot interact with Stellar assets unless that Stellar asset has a deployed SAC. Anyone can deploy the SAC for a Stellar asset.
```

However:

### TL;DR

If possible, we recommend issuing a Stellar asset and using the SAC to interact with that asset in smart contracts or to send to contract addresses. More on why below.

## Issuing assets on Stellar

Stellar has first-class support for asset tokenization — issuing an asset can be done using a [built-in transaction](https://developers.stellar.org/docs) without the development of a smart contract.

Stellar’s transactions are fast and cost-effective, making the network great for remittances and micropayments. It also has built-in features for compliance, asset management, and auditing. If you are just looking to perform simple transfers, issuing assets on Stellar has all the needed capabilities.

Stellar assets:

- Are compatible with Stellar ecosystem products (such as Stellar wallets) and other ecosystem products (such as exchanges).
- Benefit from [anchors](../../learn/encyclopedia/anchors.mdx), the bridges between the Stellar network and traditional financial systems. Explore the global [Stellar anchor directory](https://anchors.stellar.org/) for further details.

> - Give the issuer granular control over asset management with features that allow the issuer to [name the asset](../../issuing-assets/control-asset-access.mdx#naming-an-asset), [determine access control](../../issuing-assets/control-asset-access.mdx#controlling-access-to-an-asset-with-flags), [limit asset supply](../../issuing-assets/control-asset-access.mdx#limiting-the-supply-of-an-asset), [publish asset information](../../issuing-assets/publishing-asset-info.mdx), and [ensure compliance](../../issuing-assets/anatomy-of-an-asset.mdx#compliance).

Reviewer 2: These things are all possible with a custom token, it would just be more work to build it rather than using the already-implemented features of Stellar asset tokens.

### Stellar Asset Contract

The Stellar Asset Contract (SAC) is compiled into the protocol layer and allows smart contracts to interact with assets issued on Stellar. Every Stellar asset has reserved a SAC on the network that can be deployed by anyone who wants to interact with the asset from a contract. The SAC has access to all account (for XLM) and trustline (for all other assets) balances as well as smart contract token balances.

Read more about the SAC [here](./stellar-asset-contract.mdx).

Learn how to deploy a Stellar Asset Contract for an asset in [this How-To Guide](../../smart-contracts/guides/cli/deploy-stellar-asset-contract.mdx).

**Benefits of the SAC:**

- Compatibility: the SAC benefits from Stellar assets' existing interoperability.
- Cost and resource efficiency: the SAC is built into the protocol-level code that already runs the network. So, there is no need to spin up a Wasm contract’s VM to invoke a function. Each function within the SAC will be more resource-efficient than its custom-coded counterpart.
- Less work: you don’t have to write an entirely new contract. A Stellar asset’s SAC already exists on the network and just needs to be deployed to be used.

> - Customization: in addition to the customizations for Stellar assets listed above, asset issuers can set a different smart contract as an admin for their asset’s SAC. Making this admin another smart contract means that that address can add a bit of customization and logic to how the asset will authorize balances and trustlines, mint tokens, etc. Admin addresses can be contracts.

Reviewer 2: I think we can remove the preface here
```suggestion
- Customization: asset issuers can set a different smart contract as an admin for their asset’s SAC. Making this admin another smart contract means that that address can add a bit of customization and logic to how the asset will authorize balances and trustlines, mint tokens, etc. Admin addresses can be contracts.
```

**Downside of the SAC:**

- Other than the customization noted above, it is not possible to modify the behavior of Stellar assets or their SAC. If you’re looking to use assets in a way not supported by Stellar assets, you can create your own custom smart contract token using the token interface and all applications that interact with tokens using the token interface will be able to interact with the custom token.

## Custom tokens

If you have a unique use case where the SAC's capabilities are not sufficient, you can create a custom token that implements the [token interface](./token-interface.mdx). The token interface specifies the functions and events a contract must implement to be compatible with applications that use tokens.

The SAC also implements the token interface and applications that interoperate with the token interface can seamlessly interact with Stellar assets and custom tokens.

:::note

Tokens created with smart contracts cannot interact with Stellar assets unless that Stellar asset has a deployed SAC. Anyone can deploy the SAC for a Stellar asset.

> :::

Reviewer 2: Its not clear what it means to "interact" here. Can we clarify what this interaction would be?

Author: Does "transact" work? Or maybe... "Smart contracts cannot use Stellar assets unless that Stellar asset has a deployed SAC." Does that make more sense?

Reviewer 2: "Smart contracts cannot use Stellar assets unless that Stellar asset has a deployed SAC" makes a lot of sense. I think I was confused with "Tokens created with smart contracts" rather than just "smart contracts".

Author: Ok awesome, thank you!

- Benefit from [anchors](../../learn/encyclopedia/anchors.mdx), the bridges between the Stellar network and traditional financial systems. Explore the global [Stellar anchor directory](https://anchors.stellar.org/) for further details.
- Give the issuer granular control over asset management with features that allow the issuer to [name the asset](../../issuing-assets/control-asset-access.mdx#naming-an-asset), [determine access control](../../issuing-assets/control-asset-access.mdx#controlling-access-to-an-asset-with-flags), [limit asset supply](../../issuing-assets/control-asset-access.mdx#limiting-the-supply-of-an-asset), [publish asset information](../../issuing-assets/publishing-asset-info.mdx), and [ensure compliance](../../issuing-assets/anatomy-of-an-asset.mdx#compliance).

:::note

Note that while these items are also possible with custom smart contract tokens, it is more work to build the token contract rather than using the already-implemented features of Stellar asset tokens.

:::

Assets issued on the Stellar network are accessible to smart contracts with the use of that asset’s Stellar Asset Contract (SAC).

### Stellar Asset Contract

> The Stellar Asset Contract (SAC) is compiled into the protocol layer and allows smart contracts to interact with assets issued on Stellar. A single instance of the SAC can be deployed for every Stellar asset by anyone who wants to interact with the asset from a contract. The SAC has access to all account (for XLM) and trustline (for all other assets) balances as well as smart contract token balances.

Reviewer: Subtle change that changes the emphasis of the second sentence. The sentence isn't incorrect, but it flows better if the singularity is implied I think.
```suggestion
The Stellar Asset Contract (SAC) is compiled into the protocol layer and allows smart contracts to interact with assets issued on Stellar. An instance of the SAC can be deployed for every Stellar asset by anyone who wants to interact with the asset from a contract. The SAC has access to all account (for XLM) and trustline (for all other assets) balances as well as smart contract token balances.
```

> The Stellar Asset Contract (SAC) is compiled into the protocol layer and allows smart contracts to interact with assets issued on Stellar. A single instance of the SAC can be deployed for every Stellar asset by anyone who wants to interact with the asset from a contract. The SAC has access to all account (for XLM) and trustline (for all other assets) balances as well as smart contract token balances.

Reviewer: Couple of small things, missing 's's for plurals, and trust line is two words. Also, having "balances" after the brackets makes it a little awkward to read because the brackets are inserted into the thing being referenced rather than after it.
```suggestion
The Stellar Asset Contract (SAC) is compiled into the protocol layer and allows smart contracts to interact with assets issued on Stellar. A single instance of the SAC can be deployed for every Stellar asset by anyone who wants to interact with the asset from a contract. The SAC has access to all account balances (for XLM) and trust line balances (for all other assets) as well as smart contract balances.
```

Read more about the SAC [here](./stellar-asset-contract.mdx).

Learn how to deploy a Stellar Asset Contract for an asset in [this How-To Guide](../../smart-contracts/guides/cli/deploy-stellar-asset-contract.mdx).

**Benefits of the SAC:**

- Compatibility: the SAC benefits from Stellar assets' existing interoperability.

> - Cost and resource efficiency: the SAC is built into the protocol-level code that already runs the network. So, there is no need to spin up a Wasm contract’s VM to invoke a function. Each function within the SAC will be more resource-efficient than its custom-coded counterpart.

Reviewer: Abbreviated to remove some terms that don't show up often in dev vocabularies. e.g. "protocol level code."
```suggestion
- Cost and resource efficiency: the SAC is built into the protocol instead of being a contract that runs in a virtual machine. Each function within the SAC will be more resource-efficient than its custom-coded counterpart.
```

- Less work: you don’t have to write an entirely new contract. A Stellar asset’s SAC already exists on the network and just needs to be deployed to be used.

> - Customization: asset issuers can set a different smart contract as an admin for their asset’s SAC. Making this admin another smart contract means that that address can add a bit of customization and logic to how the asset will authorize balances and trustlines, mint tokens, etc. Admin addresses can be contracts.

Reviewer: Some tweaks:
```suggestion
- Customization:  Admin addresses can be contracts. Asset issuers can set a different smart contract as an admin for their asset’s SAC. Making the admin another smart contract allows the addition of custom and decentralized logic for the assets admin capabilities, such as authorizing balances and trust lines, minting tokens, etc.
```

**Downside of the SAC:**

- Other than the customization noted above, it is not possible to modify the behavior of Stellar assets or their SAC. If you’re looking to use assets in a way not supported by Stellar assets, you can create your own custom smart contract token using the token interface and all applications that interact with tokens using the token interface will be able to interact with the custom token.

## Custom tokens

> If you have a unique use case where the SAC's capabilities are not sufficient, you can create a custom token that implements the [token interface](./token-interface.mdx). The token interface specifies the functions and events a contract must implement to be compatible with applications that use tokens.

Reviewer: I think it is more accurate to say Stellar Assets here rather than the SAC. The SAC is just a way to interact with Stellar Assets, and Stellar Assets is where the capabilities are broadly.
```suggestion
If you have a unique use case where the capabilities Stellar Assets are not sufficient, you can create a custom token that implements the [token interface](./token-interface.mdx). The token interface specifies the functions and events a contract must implement to be compatible with applications that use tokens.
```

The SAC also implements the token interface and applications that interoperate with the token interface can seamlessly interact with Stellar assets and custom tokens.

:::note

> Smart contracts cannot use Stellar assets unless that Stellar asset has a deployed SAC.

Reviewer: I think the note needs to include that anyone can deploy it, otherwise this note reads like a "warning here's a blocker".
```suggestion
Smart contracts cannot use Stellar assets unless that Stellar asset has a deployed SAC. Anyone can deploy the SAC for a Stellar asset to its reserved address.
```
